vars/spinInfra.groovy

```diff
@@ -98,13 +104,47 @@ def call(Map<String, ?> params) {
         sh "terraform --version"
 
         sh """
-          terraform init -reconfigure \
-            -backend-config "storage_account_name=${env.STORE_sa_name_template}${config.subscription}" \
-            -backend-config "container_name=${env.STORE_sa_container_name_template}${environmentDeploymentTarget}" \
-            -backend-config "resource_group_name=${env.STORE_rg_name_template}-${config.subscription}" \
-            -backend-config "key=${config.productName}/${environmentDeploymentTarget}/terraform.tfstate"
+              terraform init -reconfigure \
+              -backend-config "storage_account_name=${env.STORE_sa_name_template}${config.subscription}" \
+              -backend-config "container_name=${env.STORE_sa_container_name_template}${environmentDeploymentTarget}" \
+              -backend-config "resource_group_name=${env.STORE_rg_name_template}-${config.subscription}" \
+              -backend-config "key=${config.productName}/${environmentDeploymentTarget}/terraform.tfstate"
+        """
+
+        //check tf version
+        def fmtTerraformcheck = sh(returnStatus:true, script: 'terraform fmt -check=true -recursive')
+        echo "Terraform fmt exit status ${fmtTerraformcheck}"
+          
+        if (fmtExitCode != 0) {
+            echo 'Terraform code is not formatted correctly'
+
+        // Format the Terraform code recursively
+        sh 'terraform fmt -recursive'
+
+        // Commit the formatting changes
+        git fetch origin $BRANCH:$BRANCH
+        sh  'git remote set-url origin $(git config remote.origin.url | sed "s/github.com/${USER_NAME}:${BEARER_TOKEN}@github.com/g ")'
+
+        sh "git config --global user.name ${USER_NAME}:"
```

Comment:
all these `sh` steps should be combined into one and they probably should be in an external script in the resources folder, see an example:

https://github.com/hmcts/cnp-jenkins-library/blob/49a08140fa4c1cd39b93153673cd1ad0881b22a4/vars/savePodsLogs.groovy#L16

https://github.com/hmcts/cnp-jenkins-library/blob/49a08140fa4c1cd39b93153673cd1ad0881b22a4/vars/savePodsLogs.groovy#L20